src/listeners/MainListener.js

```diff
@@ -95,6 +95,7 @@ module.exports = class MainListener extends EventListener {
 
   async onMessage (message) {
     if (message.author.bot) return
+    if (message.channel.type !== 'dm') this.modules.leveling.giveExperience(message.author.id)
```

Comment:
This function should run after the message response is sent, cause it will block the execution of the script (which is not required).

Reply:
Isn't it async?